cmake: add the `-dynamic-linker=...` form to the -dynamic-linker regex

michoecho opened the pull request:
On my system (Nix), the compiler produces a `-dynamic-linker=/nix/store/...` in the linker call scanned by get_padded_dynamic_linker_option. But the regex can't deal with the `=` there, it requires a ` `. Fix that.

Could be backported, but it's not needed for anything.

tchaikov reviewed (approved):
lgtm. but shall we apply a similar change to `configure.py`?  as it is using https://github.com/scylladb/scylladb/blob/48a4efba2f0b987bca9be2a7df978aeafbd2a2d8/configure.py#L1705 . something like
```diff
diff --git a/configure.py b/configure.py
index 4610030970..5bc7a9c8c0 100755
--- a/configure.py
+++ b/configure.py
@@ -1702,7 +1702,7 @@ def generate_version(date_stamp):
 # the program headers.
 def dynamic_linker_option():
     gcc_linker_output = subprocess.check_output(['gcc', '-###', '/dev/null', '-o', 't'], stderr=subprocess.STDOUT).decode('utf-8')
-    original_dynamic_linker = re.search('-dynamic-linker ([^ ]*)', gcc_linker_output).groups()[0]
+    original_dynamic_linker = re.search('-dynamic-linker[ =]([^ ]*)', gcc_linker_output).groups()[0]
 
     employ_ld_trickery = True
     # distro-specific setup
```
 or it is immune to this problem?

michoecho commented:
> lgtm. but shall we apply a similar change to `configure.py`? as it is using
> 
> https://github.com/scylladb/scylladb/blob/48a4efba2f0b987bca9be2a7df978aeafbd2a2d8/configure.py#L1705
> . something like
> 
> ```diff
> diff --git a/configure.py b/configure.py
> index 4610030970..5bc7a9c8c0 100755
> --- a/configure.py
> +++ b/configure.py
> @@ -1702,7 +1702,7 @@ def generate_version(date_stamp):
>  # the program headers.
>  def dynamic_linker_option():
>      gcc_linker_output = subprocess.check_output(['gcc', '-###', '/dev/null', '-o', 't'], stderr=subprocess.STDOUT).decode('utf-8')
> -    original_dynamic_linker = re.search('-dynamic-linker ([^ ]*)', gcc_linker_output).groups()[0]
> +    original_dynamic_linker = re.search('-dynamic-linker[ =]([^ ]*)', gcc_linker_output).groups()[0]
>  
>      employ_ld_trickery = True
>      # distro-specific setup
> ```
> 
> or it is immune to this problem?

If you scroll down just 2 lines from there, you will see that someone added a Nix-specific hack which completely disables the `-dynamic-linker` overwrite on Nix. So it's not that it's immune to the problem, someone just hacked around it :P

I updated the patch with a corresponding change to the regex in configure.py, and I removed the Nix-specific hack.

michoecho commented:
The Nix build scripts in the Scylla repository are unmaintained and broken anyway, so I doubt even more that anyone other than me builds Scylla on Nix. I don't use those scripts, I have my own private ones.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14442/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/13910/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/5900/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/2780/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14442/)
#### Build Details:
- Duration: 10 hr
- Builder: i-081858fb74ecdd529 (m5ad.8xlarge)

tchaikov commented:
> #### 🔴 CI State: FAILURE
> 
> ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14442/) ✅ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/13910/) ✅ - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/5900/) ✅ - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/2780/) ❌ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14442/)
> #### Build Details:
> 
>     * Duration: 10 hr
> 
>     * Builder: i-081858fb74ecdd529 (m5ad.8xlarge)

```
a boost test failed: test/boost/Debug/combined_tests --run_test=schema_change_test/test_schema_digest_does_not_change_without_digest_feature 
```

#22243

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14469/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/5916/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/2796/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/13927/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14469/)
#### Build Details:
- Duration: 10 hr
- Builder: i-08e8992f2054d91d4 (m5ad.8xlarge)

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14530/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/5966/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/13974/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/2847/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14530/)
#### Build Details:
- Duration: 10 hr
- Builder: i-04c428f99c30ceae6 (m5ad.8xlarge)

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/15847/)
:white_check_mark: - [dtest with gossip topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/15026/)
:white_check_mark: - [dtest with consistent topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/6998/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/3883/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/15847/)
#### Failed Tests (2/45008):
- [testClusteringColumnFiltering.debug.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/15847/testReport/junit/cqlpy.cassandra_tests.validation.entities/frozen_collections_test/Tests___Unit_Tests___testClusteringColumnFiltering_debug_1) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+testClusteringColumnFiltering.debug.1)
- [cqlpy.cassandra_tests/validation/entities/frozen_collections_test.debug.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/15847/testReport/junit/%28root%29/non-boost%20tests/Tests___Unit_Tests___cqlpy_cassandra_tests_validation_entities_frozen_collections_test_debug_1) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+cqlpy.cassandra_tests/validation/entities/frozen_collections_test.debug.1)
#### Build Details:
- Duration: 9 hr 11 min
- Builder: i-008399d47fc7917ed (m5ad.12xlarge)

denesb commented:
> The Nix build scripts in the Scylla repository are unmaintained and broken anyway, so I doubt even more that anyone other than me builds Scylla on Nix. I don't use those scripts, I have my own private ones.

The guy who wrote those left years ago. Maybe we should remove them, if even you don't use them.

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Framework test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/16448/)
:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/16448/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/4395/)
:white_check_mark: - [dtest with consistent topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/7533/)
:white_check_mark: - [dtest with gossip topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/15583/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/16448/)
#### Build Details:
- Duration: 7 hr 1 min
- Builder: spider1.cloudius-systems.com
